frontend/javascripts/oxalis/model/helpers/compaction/compact_toggle_actions.ts

```diff
@@ -181,5 +178,5 @@ export default function compactToggleActions(
     ...exceptions.map((tree) => updateTreeVisibility(tree)),
   ];
   const finalToggleActions = shouldUseToggleGroup ? compactedToggleActions : toggleActions;
-  return remainingActions.concat(finalToggleActions);
+  return [...remainingActions, ...finalToggleActions];
```

Comment:
I'd be careful with array spreads due to performance. Since there might be lots of update actions here, I'd stick to concat.

small microbenchmark in chrome:
```
console.time("a")
res = arr.concat(arr)
console.timeEnd("a")
console.time("a")
res = [...arr, ...arr]
console.timeEnd("a")

a: 0.332763671875 ms
a: 13.9931640625 ms
```

Reply:
arghh... 😱 

concat drives TS into a rage: 
```No overload matches this call.
  Overload 1 of 2, '(...items: ConcatArray<UpdateTreeUpdateAction | { readonly name: "deleteTree"; readonly value: { readonly id: number; }; } | { readonly name: "mergeTree"; readonly value: { readonly sourceId: number; readonly targetId: number; }; } | ... 23 more ... | { ...; }>[]): (UpdateTreeUpdateAction | ... 25 more ... | { ...; })[]', gave the following error.
    Argument of type '({ readonly name: "updateTreeVisibility"; readonly value: { readonly treeId: number; readonly isVisible: boolean; }; } | { readonly name: "updateTreeGroupVisibility"; readonly value: { readonly treeGroupId: number | ... 1 more ... | undefined; readonly isVisible: boolean; }; })[]' is not assignable to parameter of type 'ConcatArray<UpdateTreeUpdateAction | { readonly name: "deleteTree"; readonly value: { readonly id: number; }; } | { readonly name: "mergeTree"; readonly value: { readonly sourceId: number; readonly targetId: number; }; } | ... 23 more ... | { ...; }>'.
      The types returned by 'slice(...)' are incompatible between these types.
        Type '({ readonly name: "updateTreeVisibility"; readonly value: { readonly treeId: number; readonly isVisible: boolean; }; } | { readonly name: "updateTreeGroupVisibility"; readonly value: { readonly treeGroupId: number | ... 1 more ... | undefined; readonly isVisible: boolean; }; })[]' is not assignable to type '(UpdateTreeUpdateAction | { readonly name: "deleteTree"; readonly value: { readonly id: number; }; } | { readonly name: "mergeTree"; readonly value: { readonly sourceId: number; readonly targetId: number; }; } | ... 23 more ... | { ...; })[]'.
          Type '{ readonly name: "updateTreeVisibility"; readonly value: { readonly treeId: number; readonly isVisible: boolean; }; } | { readonly name: "updateTreeGroupVisibility"; readonly value: { readonly treeGroupId: number | ... 1 more ... | undefined; readonly isVisible: boolean; }; }' is not assignable to type 'UpdateTreeUpdateAction | { readonly name: "deleteTree"; readonly value: { readonly id: number; }; } | { readonly name: "mergeTree"; readonly value: { readonly sourceId: number; readonly targetId: number; }; } | ... 23 more ... | { ...; }'.
            Type '{ readonly name: "updateTreeVisibility"; readonly value: { readonly treeId: number; readonly isVisible: boolean; }; }' is not assignable to type 'UpdateTreeUpdateAction | { readonly name: "deleteTree"; readonly value: { readonly id: number; }; } | { readonly name: "mergeTree"; readonly value: { readonly sourceId: number; readonly targetId: number; }; } | ... 23 more ... | { ...; }'.
              Type '{ readonly name: "updateTreeVisibility"; readonly value: { readonly treeId: number; readonly isVisible: boolean; }; }' is not assignable to type '{ name: "mergeAgglomerate"; value: { agglomerateId1: number; agglomerateId2: number; mag: Vector3; segmentId1: number | undefined; segmentId2: number | undefined; segmentPosition1?: Vector3 | undefined; segmentPosition2?: Vector3 | undefined; }; }'.
                Types of property 'name' are incompatible.
                  Type '"updateTreeVisibility"' is not assignable to type '"mergeAgglomerate"'.
  Overload 2 of 2, '(...items: (UpdateTreeUpdateAction | { readonly name: "deleteTree"; readonly value: { readonly id: number; }; } | { readonly name: "mergeTree"; readonly value: { readonly sourceId: number; readonly targetId: number; }; } | ... 24 more ... | ConcatArray<...>)[]): (UpdateTreeUpdateAction | ... 25 more ... | { ...; })[]', gave the following error.
    Argument of type '({ readonly name: "updateTreeVisibility"; readonly value: { readonly treeId: number; readonly isVisible: boolean; }; } | { readonly name: "updateTreeGroupVisibility"; readonly value: { readonly treeGroupId: number | ... 1 more ... | undefined; readonly isVisible: boolean; }; })[]' is not assignable to parameter of type 'UpdateTreeUpdateAction | { readonly name: "deleteTree"; readonly value: { readonly id: number; }; } | { readonly name: "mergeTree"; readonly value: { readonly sourceId: number; readonly targetId: number; }; } | ... 24 more ... | ConcatArray<...>'.
      Type '({ readonly name: "updateTreeVisibility"; readonly value: { readonly treeId: number; readonly isVisible: boolean; }; } | { readonly name: "updateTreeGroupVisibility"; readonly value: { readonly treeGroupId: number | ... 1 more ... | undefined; readonly isVisible: boolean; }; })[]' is not assignable to type 'ConcatArray<UpdateTreeUpdateAction | { readonly name: "deleteTree"; readonly value: { readonly id: number; }; } | { readonly name: "mergeTree"; readonly value: { readonly sourceId: number; readonly targetId: number; }; } | ... 23 more ... | { ...; }>'.```

Reply:
does sth like this work? `return (remainingActions as UpdateAction[]).concat(finalToggleActions);`

I tested this in a playground with this:

```
const arr = [1, 2, 3]
const arr2 = [true, false];

(arr as Array<number | boolean>).concat(arr2)
```

Reply:
Yes, more or less. see PR feedback commit